addons/web/static/src/js/views/search_extension.js

```diff
@@ -0,0 +1,77 @@
+odoo.define("web/static/src/js/views/search_extension.js", function (require) {
+    "use strict";
+
+    const ActionModel = require("web/static/src/js/views/action_model.js");
+    const Domain = require("web.Domain");
+    const pyUtils = require("web.py_utils");
+
+    /**
+     * /!\ This extension is required in order to access the `query` getter
+     * => CPModel and SPModel are meaningless without it, maybe add a dependency ?
+     * @extends ActionModel.Extension
+     */
+    class SearchExtension extends ActionModel.Extension {
+
+        constructor() {
+            super(...arguments);
+            this.searchMenuTypes = this.config.searchMenuTypes || [];
+        }
+
+        get(property) {
+            switch (property) {
+                case "context": return this.config.actionContext;
+                case "domain": return this.config.actionDomain;
```

Comment:
not normalized. Also should evaluated here i think (that way get('domain') could be called separately)

Reply:
Ok it is processed in action manager to be normalized